Marsiple/Common/Network/MartianApiManager.swift

```diff
@@ -54,16 +54,20 @@ private extension MartianApiManager {
         task.resume()
     }
     
-    static func postData<DataType: Codable>(data: DataType, url: URL) {
+    static func postData<DataType: Codable>(data: DataType, url: URL, success: @escaping (String) -> Void, failure: @escaping (String) -> Void) {
         guard let json = encodeData(data: data) else { return }
         let request = setupRequest(url: url, method: HTTPRequestMethod.post, body: json)
         let session = URLSession.shared
         let task = session.dataTask(with: request) { data, response, error in
             if let error = error {
                 print(error.localizedDescription)
             }
-            if let response = response {
-                print(response)
+            if let httpResponse = response as? HTTPURLResponse {
```

Comment:
What happens if we're not able to cast the `response` as a `HTTPURLResponse`? 

Even if we know that something should always work, the code syntax here implies that we're trying to unwrap (working with an `optional`), therefore this can fail. 

We should handle the case when this fails.